jubilant/_all_any.py

```diff
@@ -50,6 +50,38 @@ def all_error(status: Status, *apps: str) -> bool:
     return _all_statuses_are('error', status, apps)
 
 
+def all_idle_agents(status: Status, *apps: str) -> bool:
+    """Report whether all unit agents in *status* (filtered to *apps* if provided) are "idle".
+
+    Unlike the other ``all_*`` and ``any_*`` helpers, this method looks at the status of each
+    Juju unit agent, not the workload's application or unit status.
+
+    Examples::
+
+        # Use the callable directly to wait for unit agents from all apps to be idle.
+        juju.wait(jubilant.all_idle_agents)
+
+        # Use a lambda to wait for unit agents only from specified apps (blog, mysql).
+        juju.wait(lambda status: jubilant.all_idle_agents(status, 'blog', 'mysql'))
+
+    Args:
+        status: The status object being tested.
+        apps: An optional list of application names. If provided, only the unit agents of these
+            applications are tested, and each must be present in ``status.apps``.
```

Comment:
The wording here made me think that if they weren't present some sort of error would be raised, rather than the result being false. What about about switching it around to something like ", and if not present in status.apps will test as non-idle"?

Reply:
Reworded. Not sure it's perfect, so feel free to comment further (will save for future PRs though).